src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
 #undef ASSERT
 #define ASSERT( x ) {}
 #endif
-	
+
 #ifndef UNUSED
 #define UNUSED( x ) ( x );
 #endif
 
-extern std::string Narrow(const wchar_t* wstr);
-extern std::string Narrow(const std::wstring& wstr);
-extern std::wstring Widen(const char* str);
-extern std::wstring Widen(const std::string& str);
 
-//	No-ops to help convert:
-//	No-ops to help convert:
-extern std::wstring Widen(const wchar_t* wstr);
-extern std::wstring Widen(const std::wstring& wstr);
-extern std::string Narrow(const char* str);
-extern std::string Narrow(const std::string& wstr);
 
-typedef std::basic_string<TCHAR> tstring;
 
 
+
+
+
+namespace ra {
+using cstring       = const char*;
+using cwstring      = const wchar_t*;
+using tstring       = std::basic_string<TCHAR>;
+
+template<typename CharT>
+using string_t = std::basic_string<CharT>;
```

Comment:
again this just adds another layer of abstraction. you just end up saying "string_t<char>" instead of "std::basic_string<char>", which is already aliased as "std::string"

Reply:
sure